pkg/model/message.go

```diff
@@ -14,6 +14,36 @@ type Messages struct {
 	Original bool
 }
 
+type MessagesSlice []Messages
+
+// Clone() returns a deep copy of MessagesSlice.
+func (m MessagesSlice) Clone() MessagesSlice {
+	msgs := make(MessagesSlice, len(m))
+
+	for i := range m {
+		msgs[i] = Messages{
+			Language: m[i].Language,
+			Messages: make([]Message, len(m[i].Messages)),
+			Original: m[i].Original,
+		}
+
+		copy(msgs[i].Messages, m[i].Messages)
+	}
+
+	return msgs
+}
+
+// GetOriginal returns a pointer to original messages.
+func (m MessagesSlice) GetOriginal() *Messages {
+	for i := range m {
+		if m[i].Original {
+			return &m[i]
+		}
+	}
+
+	return nil
+}
```

Comment:
Finding original and other messages might be more common behaviour. This also improves the consumption in your code base.

```suggestion
// SplitOriginal returns a pointer to the original and other messages.
func (m MessagesSlice) SplitOriginal() (original *Messages, others MessagesSlice) {
}
```